tests/register/test_ethernet_register.py

```diff
@@ -0,0 +1,139 @@
+    assert register.labels == test_labels
+
+    test_aux_enums = []
+    for test_enum in test_enums:
+        for key, value in test_enum.items():
+            test_dictionary = {'label': value, 'value': int(key)}
+            test_aux_enums.append(test_dictionary)
+
+    assert register.enums == test_aux_enums
+    assert register.enums_count == test_enums_count
+    assert register.cat_id == test_cat_id
+    assert register.scat_id == test_scat_id
+    assert register.internal_use == test_internal_use
+
+
+@pytest.mark.develop
+@pytest.mark.parametrize("test2_dtype", [REG_DTYPE.S8, REG_DTYPE.FLOAT, REG_DTYPE.DOMAIN, "Other type"])
+@pytest.mark.parametrize("test2_storage", [None, 1])
+def test_storage_dtype(test2_dtype, test2_storage):
+    enums_count = [{'0': 'TRIGGER_EVENT_AUTO', '1': 'TRIGGER_EVENT_FORCED'}]
+    # dtype test conditional
+    if not isinstance(test2_dtype, REG_DTYPE):
+        with pytest.raises(exc.ILValueError):
+            EthernetRegister(0x58F0, test2_dtype, REG_ACCESS.RW, "MON_CFG_SOC_TYPE",
+                             "none", "CONFIG", REG_PHY.NONE, 1, test2_storage,
+                             (-20, 20), "Monitoring trigger type", enums_count,
+                             "MONITORING", "SUB_CATEGORY_TEST", "No description (invent here)")
+
+    else:
+        register = EthernetRegister(0x58F0, test2_dtype, REG_ACCESS.RW, "MON_CFG_SOC_TYPE",
+                                    "none", "CONFIG", REG_PHY.NONE, 1, test2_storage,
+                                    (-20, 20), "Monitoring trigger type", enums_count,
+                                    "MONITORING", "SUB_CATEGORY_TEST", "No description (invent here)")
+        # storage test conditional
+        if test2_dtype in [REG_DTYPE.S8, REG_DTYPE.U8, REG_DTYPE.S16,
+                           REG_DTYPE.U16, REG_DTYPE.S32, REG_DTYPE.U32,
+                           REG_DTYPE.S64, REG_DTYPE.U64, REG_DTYPE.FLOAT]:
+            assert register.storage == test2_storage
+        else:
+            assert register.storage is None
+
+
+@pytest.mark.develop
+@pytest.mark.parametrize("test2_range", [(None, None), (-20, 20)])
+@pytest.mark.parametrize("test3_dtype", [REG_DTYPE.S8, REG_DTYPE.FLOAT])
+def test_range_ethernet_register(test2_range, test3_dtype):
+    enums_count = [{'0': 'TRIGGER_EVENT_AUTO', '1': 'TRIGGER_EVENT_FORCED'}]
+    register = EthernetRegister(0x58F0, test3_dtype, REG_ACCESS.RW, "MON_CFG_SOC_TYPE",
+                                "none", "CONFIG", REG_PHY.NONE, 1, 1,
+                                test2_range, "Monitoring trigger type", enums_count,
+                                "MONITORING", "SUB_CATEGORY_TEST", "No description (invent here)")
+
+    if test3_dtype == REG_DTYPE.FLOAT:
+        test_aux_range = (
+            float(test2_range[0]) if test2_range[0] else dtypes_ranges[test3_dtype]["min"],
+            float(test2_range[1]) if test2_range[1] else dtypes_ranges[test3_dtype]["max"],
```

Comment:
You paste the class logic here, for this reason, it will not fail. It may be better to hardcode the expected result to ensure the function does what you want. What do you think?